cartoframes/context.py

```diff
@@ -105,25 +105,24 @@ def __init__(self, base_url=None, api_key=None, creds=None, session=None,
         self.auth_client = APIKeyAuthClient(base_url=self.creds.base_url(),
                                             api_key=self.creds.key(),
                                             session=session)
+        self.auth_api_client = AuthAPIClient(base_url=self.creds.base_url(),
+                                              api_key=self.creds.key(),
+                                              session=session)
         self.sql_client = SQLClient(self.auth_client)
         self.creds.username(self.auth_client.username)
-        self._is_authenticated()
+        self._is_authenticated(self.auth_api_client)
         self.is_org = self._is_org_user()
 
         self._map_templates = {}
         self._srcdoc = None
         self._verbose = verbose
 
-    def _is_authenticated(self):
+    def _is_authenticated(self, auth_api_client):
         """Checks if credentials allow for authenticated carto access"""
-        try:
-            self.sql_client.send(
-                'select * from information_schema.tables limit 0')
-        except CartoException as err:
+        if not auth_api_client.is_valid_api_key():
```

Comment:
Why pass the auth api client as an argument instead of doing this:
```python
if not self.auth_api_client.is_valid_api_key():
    ...
```

Reply:
Oh, I tend to pass as parameters whatever is needed, but this is certainly an abuse of that style. I'll change it.